src/discord-cluster-manager/api/main.py

```diff
@@ -213,15 +222,75 @@ async def cli_auth(auth_provider: str, code: str, state: str):  # noqa: C901
     }
 
 
+async def _stream_submission_response(
+    submission_request, user_info, submission_mode_enum, bot_instance
+):
+    start_time = time.time()
+    task: asyncio.Task | None = None
+    try:
+        task = asyncio.create_task(
+            _run_submission(
+                submission_request,
+                user_info,
+                submission_mode_enum,
+                bot_instance,
+            )
+        )
+
+        while not task.done():
+            elapsed_time = time.time() - start_time
+            yield f"event: status\ndata: {json.dumps({'status': 'processing',
+                                                      'elapsed_time': round(elapsed_time, 2)}
+                                                      ,default=json_serializer)}\n\n"
+
+            try:
+                await asyncio.wait_for(asyncio.shield(task), timeout=15.0)
```

Comment:
a comment here would be nice. shield makes it so that the timeout does not affect the task itself?